recipes/libusb/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, MSBuild, tools
+import os
+
+class LibUSBConan(ConanFile):
+    name = "libusb"
+    description = "A cross-platform library to access USB devices"
+    license = "LGPL-2.1"
+    homepage = "https://github.com/libusb/libusb"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "libusb", "usb", "device")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "enable_udev": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, 'enable_udev': True, 'fPIC': True}
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _is_msvc(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "Visual Studio"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = self.name + "-" + self.version
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os != "Linux":
+            del self.options.enable_udev
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def system_requirements(self):
+        if self.settings.os == "Linux":
+            if self.options.enable_udev:
+                package_tool = tools.SystemPackageTool(conanfile=self)
+                libudev_name = ""
+                os_info = tools.OSInfo()
+                if os_info.with_apt:
+                    libudev_name = "libudev-dev"
+                elif os_info.with_yum:
+                    libudev_name = "libudev-devel"
+                elif os_info.with_zypper:
+                    libudev_name = "libudev-devel"
+                elif os_info.with_pacman:
+                    libudev_name = "libsystemd systemd"
+                else:
+                    self.output.warn("Could not install libudev: Undefined package name for current platform.")
+                    return
+                package_tool.install(packages=libudev_name, update=True)
+
+    def _build_visual_studio(self):
+        with tools.chdir(self._source_subfolder):
+            solution_file = "libusb_2017.sln"
```

Comment:
What about 2019?

Reply:
they don't offer such support. I need to re-use 2017.sln or just for MSBuild to update it.